token/core/fabtoken/driver/base.go

```diff
@@ -38,80 +40,65 @@ func (d *base) DefaultValidator(pp driver.PublicParameters) (driver.Validator, e
 
 func (d *base) newWalletService(
 	tmsConfig driver.Config,
-	binder common2.NetworkBinderService,
+	binder driver2.NetworkBinderService,
 	storageProvider identity.StorageProvider,
-	qe common.TokenVault,
+	qe driver.QueryEngine,
 	logger logging.Logger,
 	fscIdentity driver.Identity,
 	networkDefaultIdentity driver.Identity,
 	pp driver.PublicParameters,
 	ignoreRemote bool,
-) (*common.WalletService, error) {
+) (*wallet.Service, error) {
 	tmsID := tmsConfig.ID()
 
-	// Prepare roles
-	roles := identity.NewRoles()
 	deserializerManager := sig.NewMultiplexDeserializer()
 	identityDB, err := storageProvider.OpenIdentityDB(tmsID)
 	if err != nil {
 		return nil, errors.Wrapf(err, "failed to open identity db for tms [%s]", tmsID)
 	}
 	sigService := sig.NewService(deserializerManager, identityDB)
-	ip := identity.NewProvider(identityDB, sigService, binder, NewEIDRHDeserializer())
-	identityConfig, err := config2.NewIdentityConfig(tmsConfig)
+	identityProvider := identity.NewProvider(logger.Named("identity"), identityDB, sigService, binder, NewEIDRHDeserializer())
+	identityConfig, err := config.NewIdentityConfig(tmsConfig)
 	if err != nil {
 		return nil, errors.WithMessage(err, "failed to create identity config")
 	}
 
-	roleFactory := msp.NewRoleFactory(
-		logger,
-		tmsID,
-		identityConfig,         // config
-		fscIdentity,            // FSC identity
-		networkDefaultIdentity, // network default identity
-		ip,
-		ip, // signer service
-		ip, // endpoint service
-		storageProvider,
-		deserializerManager,
-		ignoreRemote,
-	)
-	role, err := roleFactory.NewWrappedX509(driver.OwnerRole, ignoreRemote)
+	// Prepare roles
+	roleFactory := role.NewFactory(logger, tmsID, identityConfig, fscIdentity, networkDefaultIdentity, identityProvider, identityProvider, identityProvider, storageProvider, deserializerManager)
```

Comment:
Don't we want to inject this role factory? (without passing the tmsID)

Reply:
A driver has the right to take decisions without receiving from outside the dependencies.